File: SafeSitePlus-main-mern/server/controllers/Weather.js
// Import fetch (use this for Node.js)
import fetch from 'node-fetch';

 

export const getAllWeatherData = async (req, res) => {
    console.log("ye chala");

    // const api_key = ee00048a45914954872163634242310;
    const city = 'London'; // Fetch from query or params
    const url =  'http://api.weatherapi.com/v1/current.json?key=ee00048a45914954872163634242310&q=London'


  try {
    const response = await fetch(url);

    if (!response.ok) {
      throw new Error(`Error: ${response.statusText}`);
    }

    const weatherData = await response.json(); // Parse the JSON from the response

    return res.status(200).json(weatherData); // Return data as a JSON response
  } catch (error) {
    return res.status(500).json({ message: error.message });
  }
};
 

File: python, Opencv, yolov8/weather.py
from fastapi import FastAPI, Query
import requests


WEATHER_API_KEY = "f1209aee709e480aacc170950251005"

def fetch_weather_forecast(city: str = "Lahore", days: int = 7):
    url = f"http://api.weatherapi.com/v1/forecast.json?key={WEATHER_API_KEY}&q={city}&days={days}"
    response = requests.get(url)
    return response.json()

